### DISCUSSION TITLE
A way to hook into the authorization process

### DISCUSSION THREAD

**Discussion by skuhnow** [directus/directus#16680] [Feature Requests]
We currently use a Directus Fork with a custom SAML Auth Provider. Now that you implemented SAML in the main branch, we looked into it, to see if we can use it, but we're missing a way to hook into the auth process to do additional checks:

* We need to check for the roles (from the IDP) if it's an Admin or a User, the User need to satisfy additional checks
* We need to set different roles (Admin or User) and we currently do a lookup with a custom external_identifier in the role collection

Not sure what would the best way. An extension or a way to define a callback to manipulate the "createUser" payload, or something else

Also https://github.com/directus/directus/discussions/11024 would be helpful

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
